### DISCUSSION TITLE
Using CSS modules with with-ant-design-less

### DISCUSSION THREAD

**Discussion by leftyfl1p** [vercel/next.js#14161] [Help]
My goal is to use css (less) modules with my custom react components along with ant design. I've cloned `with-ant-design-less` from the examples but after enabling `cssModules` in `next.config.js` ant design's css is missing, while my css (less) modules appear to load. The easiest way to observe this behavior is by changing `next.config.js` like so 
```js
module.exports = withLess({
  cssModules: true,
  lessLoaderOptions: {
    javascriptEnabled: true,
    modifyVars: themeVariables, // make your antd custom effective
  },
...
```

Upon next server restart all ant design css/less will be missing.

**Comment by leftyfl1p** [comment]
After many hours of googling I found the following workaround #8156 https://github.com/vercel/next-plugins/issues/507. Feels overkill for the issue but it works.

**Reply by franknoel** [reply]
I also used this on a project. We ended up having multiple issues like missing Webpack plugins (i.e. minifier) and polyfills, duplicated CSS, import order warning, etc.

**Reply by leftyfl1p** [reply]
What was the solution instead? I don't really *need* less, I mostly just wanted to change ant variables.

**Reply by franknoel** [reply]
We used it anyway but it was painful and I would personally argue against it. The problem with this is not less, but the css modules setup. You either need to load the whole Ant Design stylesheet and overwrite the variables, or load the Ant Design css/less modules with `babel-plugin-import`.

**Reply by a-vershinin** [reply]
@leftyfl1p Could you provide some work example in repo?  In Codesandbox or sonething else

**Reply by leftyfl1p** [reply]
@A-Vershinin for the issue or the hacky solution that I found?

**Comment by SolidZORO** [comment]
The official plugin has expired and is no longer applicable to the new version of next. js.

you can try [`next-plugin-antd-less`](https://github.com/SolidZORO/next-plugin-antd-less) , support next.js v9.5.
